examples/customer_support_workflow/main.py

```diff
@@ -1,120 +1,76 @@
-        the first for the ReActAgent and the second for the ManagerAgent.
-
-    Raises:
-        ValueError: If an invalid model is specified.
-    """
-    if model == "gpt":
-        connection = OpenAIConnection()
-        llm_react_agent = OpenAI(
-            connection=connection,
-            model=GPT_MODEL,
-            temperature=REACT_AGENT_TEMPERATURE,
-            max_tokens=MAX_TOKENS,
-            stop=["Observation:", "\nObservation:", "\n\tObservation:"],
-        )
-        llm = OpenAI(
-            connection=connection,
-            model=GPT_MODEL,
-            temperature=MANAGER_AGENT_TEMPERATURE,
-            max_tokens=MAX_TOKENS,
-        )
-    elif model == "claude":
-        connection = AnthropicConnection()
-        llm_react_agent = Anthropic(
-            connection=connection,
-            model=CLAUDE_MODEL,
-            temperature=REACT_AGENT_TEMPERATURE,
-            max_tokens=MAX_TOKENS,
-            stop=["Observation:", "\nObservation:"],
-        )
-        llm = Anthropic(
-            connection=connection,
-            model=CLAUDE_MODEL,
-            temperature=MANAGER_AGENT_TEMPERATURE,
-            max_tokens=MAX_TOKENS,
-        )
-    else:
-        raise ValueError(f"Invalid model specified: {model}")
-
-    return llm_react_agent, llm
-
-
-def main():
-    """
-    Set up and run the customer support workflow.
 
-    This function creates the necessary agents and tools, sets up the workflow,
-    and executes it with a sample input.
-    """
-    # Create LLM instances
-    llm_react_agent, llm_agent = create_llm_instances(model="gpt")
 
-    # Create a ReActAgent for handling internal bank API queries
-    agent_bank_support = ReActAgent(
-        name="Bank Support: Internal API",
-        role="customer support assistant for Internal Bank",
-        goal="help with provided customer requests",
-        llm=llm_react_agent,
-        tools=[BankApiSim(), HumanFeedbackTool()],
+def run_workflow(input: str) -> str:
+    # Create connection to OpenAI
+    connection = OpenAIConnection()
+    llm = OpenAI(
+        connection=connection,
+        model="gpt-4o-mini",
+        temperature=0.01,
     )
 
     # Create a ReActAgent for handling bank documentation queries
     agent_bank_documentation = ReActAgent(
-        name="Bank Support: Documentation",
-        role="customer support assistant for Internal Bank Documentation",
-        goal="help with provided customer requests regarding Internal Bank Documentation",
-        llm=llm_react_agent,
-        tools=[BankRAGTool(), HumanFeedbackTool()],
+        name="RAG Agent",
+        role="Customer support assistant for Internal Bank Documentation",
+        llm=llm,
+        tools=[BankRAGTool()],
     )
 
-    # Create a ManagerAgent to oversee the workflow
-    agent_manager = LinearAgentManager(llm=llm_agent)
+    # Create connection to Bank API
+    connection = HttpConnection(
+        method="POST",
```

Comment:
Not sure if all APIs would be POST, but as a simple example it make sense.
For more realistic example Tool/Agent should be able to operate with different endpoints (maybe make sense to pass swagger and so on). But this it another level of complexity.

Reply:
I was thinking about another version of this tool.


    @function_tool
    def bank_api(parameters: dict[str, str]):
        """
        An internal bank API.

        Available endpoints:
        * 'block_card' (int card_number, int pin_code)
        * 'make_transaction' (int card_number_sender, int card_number_reciever, int amoumt)
        * 'request_report' (int userId, int pin_code)

        As name of tool use - bank_api. Name of required endpoint has to be provided as parameter. 
        """
        if endpoint := parameters.get('endpoint', ""):
            return requests.post( f"http://127.0.0.1:5004/{endpoint}",
                                    json=parameters).json()['response']
        return "No endpoint chosen, no parameter with name 'endpoint' was passed."